kubernetes_deploy/production/deployment.yml

```diff
@@ -15,8 +16,16 @@ spec:
           httpGet:
             path: /ping.json
```

Comment:
*not a blocker*

I wonder if there anywhere where we could use `/healthcheck.json`. When a liveness probe fails, the container gets restarted. When a readiness probe fails, the container gets pulled out of traffic.

Is there any semantic where any of the checks in the health check necessitate the above behaviours?

Reply:
I'm not sure. This is a sample response of `/healthcheck.json`:

```json
{
  "Backend API test": {
    "response": {
      "*": {
        "status": true
      }, 
      "database": {
        "status": true
      }
    }, 
    "status": "healthy", 
    "url": "https://fox.civillegaladvice.service.gov.uk/status/healthcheck.json"
  }, 
  "disk": {
    "available_mb": 76143.25, 
    "available_percent": 75.64691463630018, 
    "status": "healthy", 
    "total_mb": 100656.12109375
  }
}
```

Let's clarify what happens when a container's liveness or readiness probe fails:

>A Pod is considered ready when all of its Containers are ready. One use of this signal is to control which Pods are used as backends for Services. When a Pod is not ready, it is removed from Service load balancers.

Given this, would tightly coupling the application's health to the health of the Backend API be the correct behaviour?

If the Backend API is down, the healthcheck will fail for all replicas, taking the service offline. Is this correct? Or should the application remain up? I believe the UI handles this situation.

Are there any cases where an application health check could fail in one replica and not another?

Is the `disk` check useful in a Kubernetes world?